bundles/CoreBundle/EventListener/AssetSanitizationListener.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\EventListener;
+
+
+use Pimcore\Event\AssetEvents;
+use Pimcore\Event\Model\ElementEventInterface;
+use Pimcore\Model\Asset;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use enshrined\svgSanitize\Sanitizer;
+
+/**
+ * @internal
+ */
+class AssetSanitizationListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [
+            AssetEvents::PRE_ADD => 'sanitizeAsset',
+            AssetEvents::PRE_UPDATE => 'sanitizeAsset',
+        ];
+    }
+
+    /**
+     * @param ElementEventInterface $e
+     */
+    public function sanitizeAsset(ElementEventInterface $e)
+    {
+        $element = $e->getElement();
+        if ($element instanceof Asset) {
+            $assetStream = $element->getStream();
+            if (isset($assetStream)) {
+                $mime = mime_content_type($assetStream);
```

Comment:
Why don't you use the Symfony Mime component anymore?

```suggestion
                $mime = MimeTypes::getDefault()->guessMimeType($element->getStream());
```

Reply:
that's because it expects a file path and i don't have the temp file until [there](https://github.com/pimcore/pimcore/blob/e28eab0d25bec6fda4cf09a6ce41fd3dcd9707c7/models/Asset.php#L744), it's 
 [called](https://github.com/pimcore/pimcore/blob/e28eab0d25bec6fda4cf09a6ce41fd3dcd9707c7/models/Asset.php#L561) after [preAdd/preUpdate](https://github.com/pimcore/pimcore/blob/e28eab0d25bec6fda4cf09a6ce41fd3dcd9707c7/models/Asset.php#L536).
Couldn't think of a better solution at the moment, beside creating the temp file "twice" or moving the events to the postAdd/postUpdate.